### PR TITLE
add space filling sobol sequence based optimizer

### PR DESCRIPTION
**Pull request by LGro**
This PR introduces a SciPy Sobol sequence based "optimizer".

### CODE REVIEW

File: blackboxopt/optimizers/space_filling.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+# SPDX-License-Identifier: Apache-2.0
+
+from typing import List
+
+try:
+    import numpy as np
+    from scipy.stats.qmc import Sobol
+except ImportError as e:
+    raise ImportError(
+        "Unable to import BOHB optimizer specific dependencies. "
```
Comment (sfalkner) [inline blackboxopt/optimizers/space_filling.py:13]: This should probably read:
```suggestion
        "Unable to import SpaceFilling optimizer specific dependencies. "
```

File: blackboxopt/optimizers/space_filling.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        search_space: The search space to optimize
+        objectives: The objectives of the optimization
+        seed: The sobol sequence is Owen scrambled and can be seeded for reproducibility
+    """
+
+    def __init__(
+        self, search_space: SearchSpace, objectives: List[Objective], seed: int = None
+    ) -> None:
+        super().__init__(search_space=search_space, objectives=objectives, seed=seed)
+        self._rng = np.random.default_rng(self.seed)
```
Comment (sfalkner) [inline blackboxopt/optimizers/space_filling.py:33]: Is there a particular reason to store the RNG here, too? We could also simply pass this instance to `Sobol`.

File: blackboxopt/optimizers/space_filling.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    raise ImportError(
+        "Unable to import BOHB optimizer specific dependencies. "
+        + "Make sure to install blackboxopt[space-fill]"
+    ) from e
+from blackboxopt.base import MultiObjectiveOptimizer, Objective, SearchSpace
+from blackboxopt.evaluation import EvaluationSpecification
+
+
+class SpaceFilling(MultiObjectiveOptimizer):
+    """Sobol sequence based space filling optimizer.
```
Comment (sfalkner) [inline blackboxopt/optimizers/space_filling.py:21]:
```suggestion
    """Sobol sequence based, space filling optimizer.
```

**Review by sfalkner — APPROVED** [review]
Thanks @LGro for this clean implementation!

File: blackboxopt/optimizers/space_filling.py (hunk 2 above)
Comment (LGro) [reply blackboxopt/optimizers/space_filling.py]: No, actually went with just passing the seed along and not initializing an RNG at all. Thanks!

**Pull request merged**